src/context/auth.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  OpenloginUserInfo,
+  LOGIN_PROVIDER_TYPE,
+} from "@web3auth/react-native-sdk";
+import Constants, { AppOwnership } from "expo-constants";
+import * as Linking from "expo-linking";
+import { router } from "expo-router";
+import { useEffect, useState, createContext, useContext } from "react";
+
+import { web3auth } from "@/lib/web3AuthService";
+
+const scheme = "wallet-app";
+const resolvedRedirectUrl =
+  Constants.appOwnership === AppOwnership.Expo ||
+  Constants.appOwnership === AppOwnership.Guest
+    ? Linking.createURL("sign-in", {})
+    : Linking.createURL("sign-in", { scheme });
+
+interface AuthContextValue {
+  signIn: (loginProvider: LOGIN_PROVIDER_TYPE) => Promise<void>;
+  signOut: () => Promise<void>;
+  userInfo: OpenloginUserInfo | undefined;
+  isLoading: boolean;
+}
+
+const AuthContext = createContext<AuthContextValue | null>(null);
+
+export function SessionProvider(props: React.PropsWithChildren) {
+  const [key, setKey] = useState<string>();
```

Comment:
Não sei como isso não está dando erro no TS, mas deveria né?

```suggestion
  const [key, setKey] = useState<string | undefined>();
```

Reply:
I add an empty string in the initial state.